atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/DbReadTable.java

```diff
@@ -62,16 +64,30 @@ public DbReadTable(ConnectionSupplier conns, DbQueryFactory queryFactory) {
 
     public ClosableIterator<AgnosticLightResultRow> getAllRows(
             Iterable<byte[]> rows, ColumnSelection columns, long ts, boolean includeValues) {
+        return getAllRows(rows, columns, ts, includeValues, null);
+    }
+
+    public ClosableIterator<AgnosticLightResultRow> getAllRows(
+            Iterable<byte[]> rows, ColumnSelection columns, long ts, boolean includeValues, Boolean reverseRows) {
         if (columns.noColumnsSelected()) {
             return ClosableIterators.emptyImmutableClosableIterator();
         } else if (isSingleton(rows)) {
             byte[] row = Iterables.getOnlyElement(rows);
             return run(queryFactory.getAllRowQuery(row, ts, columns, includeValues));
         } else {
-            return run(queryFactory.getAllRowsQuery(rows, ts, columns, includeValues));
+            FullQuery orderedQuery = addOrdering(Optional.ofNullable(reverseRows),
+                    queryFactory.getAllRowsQuery(rows, ts, columns, includeValues));
+            return run(orderedQuery);
         }
     }
 
+    private FullQuery addOrdering(Optional<Boolean> reverseRows, FullQuery query) {
```

Comment:
Could we use Optional above in `getAllRows` or just a Boolean object here? I'd prefer for both to be consistent, if that's feasible.

Perhaps even better: an enum for ordering - undefined, descending, or ascending.